namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -316,31 +316,33 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
     }
 
   private[this] val bindingCacheMu = new {}
-  private[this] var bindingCache: Map[(String, Path), Activity[NameTree[Name.Bound]]] = Map.empty
-  private[this] def getBind(ns: String, path: Path): Activity[NameTree[Name.Bound]] =
+  private[this] var bindingCache: Map[(String, Path, String), Activity[NameTree[Name.Bound]]] = Map.empty
+  private[this] def getBind(ns: String, path: Path, extraDtab: String): Activity[NameTree[Name.Bound]] =
     bindingCacheMu.synchronized {
-      val key = (ns, path)
+      val key = (ns, path, extraDtab)
       bindingCache.get(key) match {
         case Some(act) => act
         case None =>
-          val act = delegate(ns).bind(Dtab.empty, path)
+          val act = delegate(ns).bind(Dtab.read(extraDtab), path)
           bindingCache += (key -> act)
           act
       }
     }
 
-  private[this] def handleGetBind(ns: String, path: Path, req: Request): Future[Response] =
+  private[this] def handleGetBind(ns: String, path: Path, req: Request): Future[Response] = {
+    val extraDtab = req.getParam("dtab", "")
```

Comment:
`req.params.get("dtab")` will return an `Option[String]` if that's easier to work with